subsys/bluetooth/mesh/rpl.h

```diff
@@ -27,3 +27,4 @@ bool bt_mesh_rpl_check(struct bt_mesh_net_rx *rx,
 void bt_mesh_rpl_clear(void);
 void bt_mesh_rpl_update(struct bt_mesh_rpl *rpl,
 			struct bt_mesh_net_rx *rx);
+struct bt_mesh_rpl *bt_mesh_rpl_get(void);
```

Comment:
This is not a public API, don't you want users to call it directly?

Reply:
This is a good question. Should it be publicly available as API or shouldn't it?

Reply:
> This option will allow using
custom persistent storage approaches.

The choice can be given to the application.

Reply:
IMO, this should not be a user facing API. It's providing a writeable pointer to internal data in a safety critical module. Ideally, it wouldn't even expose this outside this module at all, but I suppose a `foreach()` type accessor might complicate the usage, so I won't veto it or anything. Providing an accessor function that returns the full modifiable object is the same as using an `extern` to globalize the variable, and we should stretch ourselves far to avoid doing this, as we lose all decoupling and separation of concerns on this data by exposing it.

If an application wishes to make their own storage mechanism for RPL, this should be considered a part of the mesh stack, and not part of the application, and it should use an internal API to achieve it. I'm heavily against putting anything like this in `include/`